packages/web-components/@sfx/autocomplete/test/autocomplete.test.ts

```diff
@@ -0,0 +1,72 @@
+import { expect, spy, stub } from './utils';
+import { Base } from '@sfx/base';
+import { TemplateResult } from 'lit-element';
+import Autocomplete from '../src/autocomplete';
+
+describe('Autcomplete Component', () => {
+  let autocomplete;
+  beforeEach(() => {
+    autocomplete = new Autocomplete();
+  });
+
+  describe('Constructor', () => {
+    it('should extend the Base class', () => {
+      expect(autocomplete).to.be.an.instanceof(Base);
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should add an eventListener to the window', () => {
+      spy(window, 'addEventListener');
+      autocomplete.connectedCallback();
+      expect(window.addEventListener).to.have.been.calledWith('sfx::autocomplete_received_results', autocomplete.receivedResults);
+    });
+
+    it('should call its super connectedCallback', () => {
+      const baseConnectedCallbackStub = stub(Base.prototype, 'connectedCallback');
+      autocomplete.connectedCallback();
+      expect(baseConnectedCallbackStub).to.have.been.called;
+    });
+  });
+
+  describe('disconnectedCallback', () => {
+    it('should remove eventListener from the window', () => {
+      spy(window, 'removeEventListener');
+      autocomplete.disconnectedCallback();
+      expect(window.removeEventListener).to.have.been.calledWith('sfx::autocomplete_received_results', autocomplete.receivedResults);
+    });
+
+    it('should call its super disconnectedCallback', () => {
+      const baseDisconnectedCallbackStub = stub(Base.prototype, 'disconnectedCallback');
+      autocomplete.disconnectedCallback();
+      expect(baseDisconnectedCallbackStub).to.have.been.called;
+    });
+  });
+
+  describe('Results property', () => {
+    it('should have default value of empty array', () => {
+      expect(autocomplete.results).to.be.an('array').that.is.empty;
```

Comment:
This is fine, but the following is terser:
```suggestion
      expect(autocomplete.results).to.deep.equal([]);
```